keps/2724-topology-aware-scheduling/README.md

```diff
@@ -257,6 +264,137 @@ the "rack" topology domain (there is only one such Pod, so the requirement
+
+**Example**:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: tas-example-jobset
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  replicatedJobs:
+  - name: leader
+    replicas: 1
+    template:
+      spec:
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-required-topology: cloud.provider.com/topology-block
+          spec:
+            containers:
+            - name: leader
+              image: registry.k8s.io/e2e-test-images/agnhost:2.53
+              args: ["pause"]
+  - name: workers
+    replicas: 2
+    template:
+      spec:
+        completions: 4
+        parallelism: 4
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-slice-required-topology: cloud.provider.com/topology-host
+              kueue.x-k8s.io/podset-slice-size: 4
+          spec:
+            containers:
+            - name: worker
+              image: registry.k8s.io/e2e-test-images/agnhost:2.53
+              args: ["pause"]
+```
+
+In this example there will be 8 (2 ReplicatedJob instances with 4 
+"completions" each) worker pods and we say that we want to split
+PodSet into slices with 4 pods each (so 2 slices in total) and
+each slice requires "host" topology domain.
+
+Since slice size is equal to "completions" for the Job, the end result
+is that each Job will be placed within a "host".
+
+For now PodSet Slice topology has to be required (cannot be "preferred").
+
+###### JobSet with two-level scheduling
+
+According to [Story 5](#story-5) we noticed that some users would like to
+co-locate all Jobs resulting from ReplicatedJob in JobSet in some 
+higher-level domain (like "block"), but also  co-locate each Job within
+lower-level domain like "host".
+
+To achieve that we are allowing to defined both main and slice topologies.
+
+**Example**:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: tas-example-jobset
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  replicatedJobs:
+  - name: leader
+    replicas: 1
+    template:
+      spec:
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-required-topology: cloud.provider.com/topology-block
+          spec:
+            containers:
+            - name: leader
+              image: registry.k8s.io/e2e-test-images/agnhost:2.53
+              args: ["pause"]
+  - name: workers
+    replicas: 2
+    template:
+      spec:
+        completions: 4
+        parallelism: 4
+        template:
+          metadata:
+            annotations:
+              kueue.x-k8s.io/podset-preferred-topology: cloud.provider.com/topology-block
+              kueue.x-k8s.io/podset-slice-required-topology: cloud.provider.com/topology-host
+              kueue.x-k8s.io/podset-slice-size: 4
```

Comment:
Same question about handling JobSets without this annotation, but with `kueue.x-k8s.io/podset-slice-required-topology`